testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/statistics/xa/XaDataSourcePoolStatisticsTestCase.java

```diff
@@ -75,13 +81,25 @@
 
     @Deployment
     public static Archive<?> deploy() {
+        // This ugly hack is needed because of the following line on TxTestUtil
+        // return CDI.current().select(TransactionCheckerSingleton.class).get();
+        final String jbossHome = System.getProperty("jboss.dist");
+        final String modulesDir = jbossHome + "/modules/system/layers/base/org/jboss/as/weld/main/";
+        final File weldModuleFile = Paths.get(modulesDir).toFile().listFiles(pathname ->
+                pathname.getName().startsWith("wildfly-weld")
+                        && pathname.getName().endsWith("-SNAPSHOT.jar"))[0];
```

Comment:
Can TxTestUtil.getChecker() be eliminated? I may have missed something but it seems like all the calls to it lead back to EJB implementations. So can those just have the TransactionCheckerSingleton injected and then pass it to the overloaded TxTestUtil methods that take one as a param?